LayoutTests/http/tests/notifications/notification-permission-query.html

```diff
@@ -0,0 +1,73 @@
+<!DOCTYPE html>
+<html>
+<head>
+<script src="/resources/js-test-pre.js"></script>
+<script src="/resources/notifications-test-pre.js"></script>
+</head>
+<body>
+<script>
+
+/*
+ * This test tests that Permissions::query() retrieves the true permission state of the Notifications API
+ * in all 3 cases: no permission state has been set (default), the permission state is granted, and the
+ * permission state is denied. 
+ */
+
+description("This test checks that Permissions::query() returns the true permission state of the Notifications API.")
+
+jsTestIsAsync = true;
+
+function defaultTest()
+{
+    testRunner.removeAllWebNotificationPermissions();
+    debug("No permission state is set");
+    navigator.permissions.query({ name: "notifications" }).then((status)=>{
+        window.permissionsQueryResult = status.state;
+        shouldBeEqualToString("permissionsQueryResult", "prompt");
+        
+        grantTest();
+    });
+}
+
+function grantTest() 
+{    
+    internals.withUserGesture(() => {
+        testRunner.grantWebNotificationPermission(testURL);
+        debug("Permission has been granted");
+        window.Notification.requestPermission().then((result)=> {
+            window.requestPermissionResult = result;
+            shouldBeEqualToString("requestPermissionResult", "granted");
+            navigator.permissions.query({ name: "notifications" }).then((status)=>{
+                window.permissionsQueryResult = status.state;
+                shouldBeEqualToString("permissionsQueryResult", "granted");
+
+                denyTest();
+            });
+        }); 
+    });
+}
+
+function denyTest() 
+{
+    internals.withUserGesture(() => {
+        testRunner.denyWebNotificationPermission(testURL);
```

Comment:
I think we should call the OnPrompt function here. Looks like we're testing the user granting / denying from the prompt.

Reply:
Fixed!